package/base-files/files/lib/functions/system.sh

```diff
@@ -79,6 +79,26 @@ mtd_get_mac_ascii() {
 	[ -n "$mac_dirty" ] && macaddr_canonicalize "$mac_dirty"
 }
 
+mtd_get_mac_encrypted_arcadyan() {
+	local mtdname="$1"
+	local part
+	local mac_dirty
+
+	part=$(find_mtd_part "$mtdname")
+	if [ -z "$part" ]; then
+		echo "mtd_get_mac_encrypted_arcadyan: partition $mtdname not found!" >&2
+		return
+	fi
+
+	# config decryption and getting mac
+	mac_dirty=$(dd if="$part" bs=1 count=$((0x170)) skip=$((0x100)) 2>/dev/null | \
+		uencrypt -d -n -k 2A4B303D7644395C3B2B7053553C5200 \
```

Comment:
I suggest to check if uencrypt is available; use it if found.  If not, check for the openssl util, which can be used instead.  I'd give uencrypt the preference, since it should run a little faster, using less memory.

Reply:
@cotequeiroz nice idea. Done.